platform-hub-api/app/models/kubernetes_cluster.rb

```diff
@@ -0,0 +1,43 @@
+class KubernetesCluster < ApplicationRecord
+
+  include FriendlyId
+  include Audited
+
+  before_save :downcase_name
+
+  audited descriptor_field: :name
+
+  friendly_id :name
```

Comment:
We don't restrict what goes in the `name` field, do we? We can either add a custom validation for this ([like we do for the `ContactList` model](https://github.com/UKHomeOffice/platform-hub/blob/d465ec5f59de446f91f8cc7a9a853de9de2363fb/platform-hub-api/app/models/contact_list.rb#L3)), OR we should do the following instead:

```ruby
# In migration, add:
t.string :slug, null: false
t.index :slug, unique: true

# Then, in model:
friendly_id :name, :use => :slugged
```

… in either case, we still need the below (since name can be edited?):

```ruby
def should_generate_new_friendly_id?
  name_changed? || super
end
```